internal/signerverifier/ed25519/ed25519.go

```diff
@@ -0,0 +1,120 @@
+package ed25519
+
+import (
+	"context"
+	"crypto"
+	"crypto/ed25519"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/adityasaky/gittuf/internal/signerverifier"
+	"github.com/adityasaky/gittuf/internal/tuf"
+	"github.com/secure-systems-lab/go-securesystemslib/cjson"
+)
+
+type Ed25519SignerVerifier struct {
+	keyID   string
+	private ed25519.PrivateKey
+	public  ed25519.PublicKey
+}
+
+// NewSignerVerifierFromSecureSystemsLibFormat creates an Ed25519SignerVerifier
+// from a key serialized in the custom securesystemslib format.
+func NewSignerVerifierFromSecureSystemsLibFormat(keyContents []byte) (*Ed25519SignerVerifier, error) {
+	var k interface{}
+	if err := json.Unmarshal(keyContents, &k); err != nil {
+		return &Ed25519SignerVerifier{}, err
+	}
+
+	t := k.(map[string]interface{})
+
+	var keyID string
+	if _, ok := t["keyid"]; ok {
+		keyID = t["keyid"].(string)
+	} else {
+		// FIXME: repetition
+		data, err := cjson.EncodeCanonical(k)
+		if err != nil {
+			panic(fmt.Errorf("error creating key ID: %w", err))
+		}
+		digest := sha256.Sum256(data)
+		keyID = hex.EncodeToString(digest[:])
+	}
+
+	keyval := t["keyval"].(map[string]interface{})
+
+	publicS := keyval["public"].(string)
+	public, err := hex.DecodeString(publicS)
+	if err != nil {
+		return &Ed25519SignerVerifier{}, err
+	}
+
+	if _, ok := keyval["private"]; ok {
+		privateS := keyval["private"].(string)
+		private, err := hex.DecodeString(privateS)
+		if err != nil {
+			return &Ed25519SignerVerifier{}, err
+		}
+		if len(private) == ed25519.PrivateKeySize/2 {
```

Comment:
As someone who's less familiar with ed25519, why do we need to do this?

Reply:
Yep, I really should add a comment there!

Reply:
Done!